spec.bs

```diff
@@ -958,6 +959,125 @@ For now, see [https://github.com/wicg/observable#operators](https://github.com/w
+          :: 1. If |activeInnerAbortController| is not null, then [=AbortController/signal abort=]
+                |activeInnerAbortController|.
+
+                Note: This "unsubscribes" from the "inner" {{Observable}} that was derived from the
+                value that was <i>last</i> pushed from |sourceObservable|. Then we immediately
+                subscribe to the <i>new</i> {{Observable}} that we're about to derive from |value|,
+                i.e., the <i>most-recently</i> pushed value from |sourceObservable|.
+
+             1. Set |activeInnerAbortController| to a [=new=] {{AbortController}}.
+
+             1. Run the [=switchmap process next value steps=] with |value|, |subscriber|, |mapper|,
+                and <b>references</b> to all of the following: |activeInnerAbortController|, and
+                |outerSubscriptionHasCompleted|.
+
+                Note: The [=switchmap process next value steps=] will subscribe to the
+                {{Observable}} derived from |value| (if one such can be derived) and keep processing
+                values from it until either (1) its subscription becomes inactive (either by error or
+                completion), or (2) |activeInnerAbortController| <a href=#switchmap-next-steps>gets
+                aborted</a>, due to |sourceObservable| pushing another <i>newer</i> value that will
+                replace the current "inner" subscription.
+
+          : [=internal observer/error steps=]
+          :: Run |subscriber|'s {{Subscriber/error()}} method, given the passed in <var
+             ignore>error</var>.
+
+          : <dt id=switchmap-complete-steps>[=internal observer/complete steps=]</dt>
+          :: 1. Set |outerSubscriptionHasCompleted| to true.
+
+                Note: If |activeInnerAbortController| is not null, then we don't immediately
+                complete |subscriber|. Instead, the [=switchmap process next value steps=] will
+                complete |subscriber| when the inner subscription finally becomes inactive on its
+                own volition.
+
+             1. If |activeInnerAbortController| is null, run |subscriber|'s
+                {{Subscriber/complete()}} method.
+
+       1. Let |options| be a new {{SubscribeOptions}} whose {{SubscribeOptions/signal}} is
+          |subscriber|'s [=Subscriber/signal=].
+
+       1. <a for=Observable lt="subscribe to an Observable">Subscribe</a> to |sourceObservable|
+          given |sourceObserver| and |options|.
+
+    1. Return |observable|.
+</div>
+
+<div algorithm>
+  The <dfn>switchmap process next value steps</dfn>, given an {{any}} |value|, a {{Subscriber}}
+  |subscriber|, a {{Mapper}} |mapper|, and <b>references</b> to all of the following: an
+  {{AbortController}} |activeInnerAbortController|, and a [=boolean=]
+  |outerSubscriptionHasCompleted|, are to run these steps:
+
+    1. Let |mappedResult| be the result of [=invoking=] |mapper| with |value|.
```

Comment:
With a value and index. All of the mapping functions should take an index as a second argument. Just like `array.map((value, index) => 'result')`.

Reply:
Looks like `flatMap()` is also missing this (although the implementation works). Filed https://github.com/WICG/observable/issues/139 for that. I'll fix `switchMap()` here.

Reply:
Done.